engine/src/move_generation.rs

```diff
@@ -676,4 +706,13 @@ mod tests {
         gs.board.black_knights = 1 << 26;
         assert!(gs.position_under_attack(9, true), "Knight not attacking");
     }
+
+    #[test]
+    fn test_move_piece() {
+        let mut gs = GameState::new();
+
+        gs.move_piece(12, 28);
+        assert_eq!(gs.board.piece_at_position(12), None);
+        assert_eq!(gs.board.piece_at_position(28), Some('P'));
+    }
```

Comment:
I feel like we should also have tests for legal moving.

Reply:
Added